x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateConfig.java

```diff
@@ -0,0 +1,34 @@
+package org.elasticsearch.xpack.core.template;
+
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+public class IndexTemplateConfig {
+
+    private final String templateName;
+    private final String fileName;
+    private final String version;
+    private final String versionProperty;
+
+    public IndexTemplateConfig(String templateName, String fileName, String version, String versionProperty) {
+        this.templateName = templateName;
+        this.fileName = fileName;
+        this.version = version;
+        this.versionProperty = versionProperty;
+    }
+
+    public String getFileName() {
+        return fileName;
+    }
+
+    public String getTemplateName() {
+        return templateName;
+    }
+
+    public byte[] load() {
+        String template = TemplateUtils.loadTemplate("/" + fileName + ".json", version,
+            Pattern.quote("${" + versionProperty + "}"));
+        assert template != null && template.length() > 0;
+        return template.getBytes(StandardCharsets.UTF_8);
```

Comment:
I wonder if we should just return the `String` here, and let the caller deal with converting it to bytes? I don't think having the `getBytes` here buys us much

Reply:
I've changed this to `loadBytes()` but given this is called currently only ever immediately fed into a `PutIndexTemplateRequest`, which needs a byte array, I'm inclined to leave it and let whoever needs it add a `loadString()` method if it's ever needed.

If you feel strongly about this I'm willing to change it, but otherwise I'll leave it returning the byte array.

Reply:
I don't feel strongly about it, mostly just curious :)